bindings/mqtt/mqtt.go

```diff
@@ -311,3 +311,10 @@ func (m *MQTT) createClientOptions(uri *url.URL, clientID string) *mqtt.ClientOp
 
 	return opts
 }
+
+func (m *MQTT) Close() error {
+	m.consumer.Disconnect(1)
```

Comment:
I believe it is possible for `m.consumer` to be `nil` because it is initialized in `Read` and not `Init`. Because of that you will want an `if m.consumer != nil { m.consumer.Disconnect(1) }`.

Reply:
Done.